airflow-core/src/airflow/models/connection.py

```diff
@@ -488,6 +488,10 @@ def get_connection_from_secrets(cls, conn_id: str) -> Connection:
         :param conn_id: connection id
         :return: connection
         """
+        # Debuggers and other tooling probe dunder attributes like "__iter__". These should never be treated
+        # as real connection IDs.
+        if conn_id.startswith("__"):
+            raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined")
```

Comment:
I don’t think we can simply do this since it has compatibility implications. Why do those names get passed in here as a string in the first place?